tests/test_cli_compile.py

```diff
@@ -684,8 +684,8 @@ def test_url_package(runner, line, dependency, generate_hashes):
         ["-n", "--rebuild", "--no-build-isolation"]
         + (["--generate-hashes"] if generate_hashes else []),
     )
-    assert out.exit_code == 0
     assert dependency in out.stderr
+    assert out.exit_code == 0
```

Comment:
I'd prefer not to include unrelated changes right in this PR. But if you keep this, it's probably better to merge the instructions into a single `assert` statement. This way, it'd be following the [AAA principle](https://docs.pytest.org/en/stable/explanation/anatomy.html) to the letter: one test == one assertion.
```suggestion
    assert 0 == out.exit_code and dependency in out.stderr
```